recipes/pcapplusplus/all/conanfile.py

```diff
@@ -34,19 +38,34 @@ def config_options(self):
             del self.options.fPIC
 
     def requirements(self):
-        if self.settings.os != "Windows":
+        if self.settings.os == "Windows":
+            self.requires("pthreads4w/3.0.0")
+            self.requires("npcap/1.70")
+        else:
             self.requires("libpcap/1.9.1")
 
+    def _get_vs_version(self):
+        if not is_msvc(self):
+            return None
+        vs_mapping = {
+            "14": "vs2015",
+            "15": "vs2017",
+            "16": "vs2019",
+            # configure-windows-visual-studio.bat does not know vs2022
+            # we use vs2019 and change PlatformToolset later
+            "17": "vs2019",
```

Comment:
I wouldn't recommend supporting vs2022 since PcapPlusPlus doesn't officially support it yet. We can add it later when PcapPlusPlus officially supports it

Reply:
As far as I can tell there is no logic on vs version in the source code, only in `configure-windows-visual-studio.bat`
and compatibility between different versions of vs is ensured since vs2015 as long as `/GL` and `/LTCG` flags are not used.

See https://learn.microsoft.com/en-us/cpp/porting/binary-compat-2015-2017?view=msvc-170

Do you have a date for when the next version will be released with vs2022 support? vs2022 has be released for almost a year now (November 2021)

Reply:
I really hope we can release a new version before the end of this year, hopefully replacing the current build system with [CMake](https://github.com/seladb/PcapPlusPlus/pull/944), which should make supporting VS 2022 much easier. 

Although VS 2022 is compatible with previous versions, it'll look a bit weird that PcapPlusPlus doesn't officially support VS 2022, but the Conan package does... what do you think?

Reply:
I think that "not supported" doesn't mean it cannot work or binaries weren't generated, it just means it was never tested and there is no guarantee it will work.

Also, in PcapPlusPlus case, the configuration script [`configure-windows-visual-studio.bat`](https://github.com/seladb/PcapPlusPlus/blob/master/configure-windows-visual-studio.bat) specifically asks for the VS version and VS 2022 isn't included. 

One can say the same thing about other platforms, for example iOS - maybe we can get it to work on iOS but I didn't try so I don't know...

The PcapPlusPlus web-site mentions the supported platforms and specifically the supported VS versions:
https://pcapplusplus.github.io/docs/platforms#-windows

Having said all of that - I'll leave it to you to decide whether the package should support VS 2022 or not

Reply:
I double checked just in case, but the configuration script `configure-windows-visual-studio.bat` is just a wrapper to set `PlatformToolset` based on vs version, so simply using `vs2015` and letting conan specify PlatformToolset to vs (through MSBuild class) works as expected.

Compatibility is ensured by Microsoft/Visual Studio as described in https://learn.microsoft.com/en-us/cpp/porting/binary-compat-2015-2017?view=msvc-170